frontend/src/common/EditableText.tsx

```diff
@@ -1,34 +1,53 @@
 import EditIcon from '@mui/icons-material/Edit'
 import { Box, Button, IconButton, Stack, TextField, Tooltip, Typography } from '@mui/material'
-import { useState } from 'react'
+import { useCallback, useState } from 'react'
+import RichTextEditor from 'src/common/RichTextEditor'
 
 interface EditableTextProps {
   value: string
   onSubmit: (newValue: string) => void
   tooltipText?: string
   submitButtonText?: string
+  multiline?: boolean
+  richText?: boolean
 }
 
 export default function EditableText({
   value,
   onSubmit,
   tooltipText = 'Edit this text',
   submitButtonText = 'Submit',
+  multiline = false,
+  richText = false,
 }: EditableTextProps) {
   const [isEditMode, setIsEditMode] = useState(false)
   const [newValue, setNewValue] = useState(value)
 
   const previousText = value
 
-  function handleCancelOnClick() {
+  const handleCancelOnClick = useCallback(() => {
     setIsEditMode(false)
     setNewValue(previousText)
-  }
+  }, [previousText])
 
-  function handleSubmit() {
+  const handleSubmit = useCallback(() => {
     setIsEditMode(false)
     onSubmit(newValue)
+  }, [newValue, onSubmit])
+
+  const submitButtons = () => {
```

Comment:
I'd suggest using useMemo here so we don't have to call this as a function in our JSX below